.phpcs.xml

```diff
@@ -22,6 +22,205 @@
+		<exclude name="MediaWiki.Usage.ForbiddenFunctions.assert" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.NotParenthesisParamType" />
+		<exclude name="Squiz.WhiteSpace.ObjectOperatorSpacing.After" />
+		<exclude name="Squiz.Classes.SelfMemberReference.SpaceAfter" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.SpacingBeforeReturnType" />
+		<exclude name="PSR2.ControlStructures.SwitchDeclaration.SpaceBeforeColonCASE" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.DuplicateReturn" />
+		<exclude name="Universal.UseStatements.KeywordSpacing.SpaceBeforeAs" />
+		<exclude name="PSR2.Methods.MethodDeclaration.StaticBeforeVisibility" />
+		<exclude name="Generic.NamingConventions.ConstructorName.OldStyle" />
+		<exclude name="Squiz.Scope.MethodScope.Missing" />
+		<exclude name="Generic.NamingConventions.UpperCaseConstantName.ClassConstantNotUpperCase" />
+		<exclude name="Squiz.Scope.MethodScope.Missing" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.UppercasePrimitiveStringParam" />
+		<exclude name="MediaWiki.Usage.ExtendClassUsage.FunctionVarUsage" />
+		<exclude name="MediaWiki.Commenting.PropertyDocumentation.NotPunctuationVarType" />
+		<exclude name="MediaWiki.WhiteSpace.SpaceBeforeBracket.SpaceFound" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.SpacingAfterReturnType" />
+		<exclude name="MediaWiki.Usage.MagicConstantClosure.FoundConstantMethod" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.UppercasePrimitiveCallableReturn" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.SpacingAfter" />
+		<exclude name="Generic.PHP.LowerCaseConstant.Found" />
+		<exclude name="MediaWiki.Commenting.PropertyDocumentation.NotShortIntArrayVar" />
+		<exclude name="PSR2.ControlStructures.SwitchDeclaration.SpacingAfterCase" />
+		<exclude name="MediaWiki.WhiteSpace.OpeningKeywordParenthesis.WrongWhitespaceBeforeParenthesis" />
+		<exclude name="PSR2.ControlStructures.SwitchDeclaration.BodyOnNextLineDEFAULT" />
+		<exclude name="PSR2.Methods.MethodDeclaration.Underscore" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.MissingReturnType" />
+		<exclude name="PSR2.Classes.PropertyDeclaration.StaticBeforeVisibility" />
+		<exclude name="MediaWiki.Usage.ExtendClassUsage.FunctionConfigUsage" />
+		<exclude name="MediaWiki.Commenting.DocComment.SpacingSingleLineCloseTag" />
+		<exclude name="MediaWiki.Commenting.DocComment.SpacingDocStarSingleLine" />
+		<exclude name="MediaWiki.Commenting.FunctionComment.UppercasePrimitiveStringReturn" />
+		<exclude name="MediaWiki.NamingConventions.PrefixedGlobalFunctions.allowedPrefix" />
+		<exclude name="MediaWiki.Commenting.LicenseComment.LicenseTagNonFileComment" />
+		<exclude name="Squiz.Functions.FunctionDeclarationArgumentSpacing.SpaceAfterEquals" />
+		<exclude name="MediaWiki.VariableAnalysis.MisleadingGlobalNames.Misleading$wgCategoryCollation" />
+		<exclude name="MediaWiki.Usage.DirUsage.FunctionFound" />
```

Comment:
:eyes: I've never seen this many excludes

Reply:
I didn't expect this number of sniffs as well. But when you enable phpcs and run **make ci**, phpcs will reported all of those sniffs. 
Inside SMW are more then 2000 files which are checked by phpcs.